rbx_binary/src/text_deserializer.rs

```diff
@@ -222,6 +222,7 @@ pub enum DecodedValues {
     Int64(Vec<i64>),
     SharedString(Vec<u32>), // For the text deserializer, we only show the index in the shared string array.
     OptionalCFrame(Vec<Option<CFrame>>),
+    UniqueId(Vec<String>),
```

Comment:
I like the decision to format these as hex strings instead of structs

Reply:
I would love some help doing this, because I'm not entirely sure what you mean. Manual implementations of Serde aren't something I bother with very often, so they're still confusing to me.

Reply:
Sure, just need something like this at the end of the unique id file:
```rust
#[cfg(feature = "serde")]
mod serde_impl {
    use super::*;

    use serde::{Deserialize, Deserializer, Serialize, Serializer};

    impl Serialize for UniqueId {
        fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
            if serializer.is_human_readable() {
		// Maybe we should just impl Display like this and then call to_string here?
                serializer.serialize_str(&format!(
                    "{:08x}{:08x}{:016x}",
                    self.index, self.time, self.random
                ))
            } else {
		// This is probably slightly wrong lol, but you should get the idea
		// Maybe this should be a method on UniqueId?
                let index = i128::from(self.index());
                let time = i128::from(self.time()) >> 32;
                let random = i128::from(self.random()) >> 64;
                serializer.serialize_i128(index | time | random)
            }
        }
    }

    impl<'de> Deserialize<'de> for UniqueId {
        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
        where
            D: Deserializer<'de>,
        {
            todo!("Dekko please add deserialize impl")
        }
    }
}

#[cfg(all(test, feature = "serde"))]
mod serde_tests {
    use super::*;

    #[test]
    fn dekko_please_add_serde_tests() {
        todo!()
    }
}
```
You can look at some of the other types (like Axes) for examples